services/generator/service.go

```diff
@@ -682,3 +682,8 @@ func (s *service) RenderToYAML(v interface{}, data interface{}) string {
 
 	return renderedTemplate.String()
 }
+
+// Adding this function to normalize the nginx configuration snippet as .estafette.yaml does not support $
+func normalizeNginxConfigurationSnippet(in string) string {
```

Comment:
We missed the scenario where individuals who forget to include `$` then will encounter deployment failures, and the regex won't let any modifications for developer needs to add nginx settings

**[Example:](https://go.dev/play/p/wjzhDE7no3h)**
```go
func normalizeNginxConfigurationSnippet(input string) string {
	// Define a regular expression pattern to match "${string}"
	pattern := "\\$\\{([a-zA-Z0-9_-]+)\\}"

	// Compile the regular expression pattern
	regex, err := regexp.Compile(pattern)
	if err != nil {
		// log the error
		fmt.Printf("Error compiling regex pattern: %v\n", err)

		return input // return the original input string unchanged
	}

	// Replace all occurrences of the pattern with "$" followed by the captured group
	return regex.ReplaceAllStringFunc(input, func(match string) string {
		// Extract the captured group (string inside {})
		variable := match[2 : len(match)-1]
		// Return "$" followed by the captured group
		return "$" + variable
	})
}
func main() {
	s := `if (${args}~ track=canary){
  set ${proxy_upstream_name}"frontend-acceptance-edge-api-canary-https";
  set ${proxy_host}{DOLLAR}proxy_upstream_name;
  set ${service_name}"edge-api-canary";
}`
	result := normalizeNginxConfigurationSnippet(s)
	fmt.Println(result)
}
```